### ISSUE TITLE
:test:fixtures:krb5kdc-fixture:composeUp failure

### ISSUE THREAD

**Issue by gwbrown** [elastic/elasticsearch#48027]
CI failed on 7.x with this error:
```
ERROR: for a5c3c1596d62b7b1842fcb2469c0c711_krb5kdc-fixture__hdfs_1  Cannot create container for service hdfs: error creating overlay mount to /var/lib/docker/overlay2/e40001620e6f9a60ea6001580ccd075835a2d8234c75c50bf056f66d9ba1cba7/merged: device or resource busy
```
[Build scan](https://gradle-enterprise.elastic.co/s/kbju23a4ponxc/console-log?task=:test:fixtures:krb5kdc-fixture:composeUp#L9)

Looks like docker-compose had trouble creating an `hdfs` container. I'm not sure if this issue is more appropriate for Elasticsearch or Infra.

**Comment by rjernst** [comment]
This seems like an issue with the underlying docker image we are trying to start. @jbaiera can you have a look?

**Comment by atorok** [comment]
This particular failure is on ubuntu-1804 I know we'we had similar issues  on SLES and maybe fedora 29 and we pulled them from the general worker pool, but didn't get to the bottom of the issue.

**Comment by droberts195** [comment]
Another one happened in https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+7.x+multijob+fast+part2/1688/consoleText:

```
Creating network "a5c3c1596d62b7b1842fcb2469c0c711_krb5kdc-fixture__default" with the default driver
Creating a5c3c1596d62b7b1842fcb2469c0c711_krb5kdc-fixture__hdfs_1 ... 
Creating a5c3c1596d62b7b1842fcb2469c0c711_krb5kdc-fixture__peppa_1 ... 
Creating a5c3c1596d62b7b1842fcb2469c0c711_krb5kdc-fixture__peppa_1 ... error

ERROR: for a5c3c1596d62b7b1842fcb2469c0c711_krb5kdc-fixture__peppa_1  Cannot create container for service peppa: error creating overlay mount to /var/lib/docker/overlay2/9d2967ca570b8c46643a30a2e801bcdde60e0dfbbed40c7f41936e41d75f6f57/merged: device or resource busy
Creating a5c3c1596d62b7b1842fcb2469c0c711_krb5kdc-fixture__hdfs_1  ... done

ERROR: for peppa  Cannot create container for service peppa: error creating overlay mount to /var/lib/docker/overlay2/9d2967ca570b8c46643a30a2e801bcdde60e0dfbbed40c7f41936e41d75f6f57/merged: device or resource busy
Encountered errors while bringing up the project.
Stopping a5c3c1596d62b7b1842fcb2469c0c711_krb5kdc-fixture__hdfs_1 ... 
```

The worker this happened on was elasticsearch-ci-immutable-ubuntu-1804-1571225771493903523.  As it was ephemeral it's gone now.

**Comment by romseygeek** [comment]
This also just happened on a CI build on one of my PRs: https://gradle-enterprise.elastic.co/s/aexbtvi3nivxo/console-log?task=:test:fixtures:krb5kdc-fixture:composeUp

**Comment by mark-vieira** [comment]
Definitely seems specific to ubuntu 18.04.

https://gradle-enterprise.elastic.co/scans/failures?failures.failureClassification=non_verification&failures.failureMessage=*%0A%3E%20Exit-code%201%20when%20calling%20/usr/local/bin/docker-compose,%20stdout:%20N/A*&list.offset=0&list.size=50&list.sortColumn=startTime&list.sortOrder=desc&search.buildToolType=gradle&search.buildToolType=maven&search.startTimeMax=1571342751552&search.startTimeMin=1570737951552&trends.section=overview&trends.timeResolution=day&viewer.tzOffset=-420

This [Debian 10 build](https://gradle-enterprise.elastic.co/s/5kxkv5aawkn3k/console-log?task=:x-pack:test:idp-fixture:composeUp) also failed for the same reason, which would make sense since Ubuntu 18.04 is based on Debian 10.

**Comment by atorok** [comment]
New failures should have a "GCP upload" link in  the build scan. We have to wait for one of those to get additional insight on this.

**Comment by mark-vieira** [comment]
@atorok I just took a look at the logs uploaded to GCP for [this failure](https://gradle-enterprise.elastic.co/s/7xaexpy2rde2k/) and the only thing in there was the Gradle daemon log which is effectively identical to the console output. What kind of Docker-specific logging were we expecting to capture here?

**Comment by mark-vieira** [comment]
@atorok it looks like we aren't getting a `journalctl` log because we need to call `supply()` here: https://github.com/elastic/elasticsearch/blob/28b599de3d7d01c195f33fff2450327ee2a66afe/gradle/build-complete.gradle#L28

**Comment by atorok** [comment]
I opened #48276  to address that

**Comment by romseygeek** [comment]
Another failure here: https://gradle-enterprise.elastic.co/s/iaherl5mpkitc

**Comment by mark-vieira** [comment]
After discussing, since this looks to be something somewhat related Docker volumes, would be to use a different storage driver here. We've already switched to using `devicemaper` on Centos 8 due to another issue, perhaps we could try using `devicemaper` across the board.

@fatmcgav What would be the implications of ☝️? In https://github.com/elastic/infra/pull/15395 we conditionally apply this to only one image, what would be involve in doing this by default? What else would this effect (ideally only elasticsearch-ci images)?

**Comment by fatmcgav** [comment]
@mark-vieira As to wider implications, I'm not sure... I'll have a look at what options exist... 

With regards to making the change across the board for `elasticsearch-ci` images, that's also something we could look at doing... We'd need to handle the `sysv`/`systemd` behaviour differences, but we've already got `systemd` supported via https://github.com/elastic/infra/pull/15395

**Comment by mark-vieira** [comment]
@fatmcgav Makes sense. My main concern is that changes in those ansible scripts are "global" and would apply to other project CI images as well.

**Comment by atorok** [comment]
We'll get a full system log, including the kernel log once elastic/infra#15520 merge. We might have it for some platforms already in the additional logs linked from the build scan. 

I think @dliappis brings up a good point. We need to make this change for elasticsearch-ci only and not packaging-ci, we have packaging tests that run on GCP these days, not only the vagrant ones.

**Comment by mark-vieira** [comment]
The `journalctl` logs haven't provided any insight. On SLES the log is completely empty and on Ubuntu there are no entries.

```
-- Logs begin at Tue 2019-10-29 04:09:52 UTC, end at Tue 2019-10-29 04:42:35 UTC. --
-- No entries --
```

I'm also staring to doubt if this is necessarily related to the storage driver as on SLES it's trying to [create a `tmpfs` mount](https://gradle-enterprise.elastic.co/s/wk3e3borma3f6/console-log?task=:test:fixtures:krb5kdc-fixture:composeUp#L9), not overlay2, yet encountering the same error.

**Comment by bizybot** [comment]
I saw this today and also on one of my PRs:
https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+multijob-unix-compatibility/os=sles-12&&immutable/360/console

**Comment by atorok** [comment]
It seems that the new images where the logs would be included were not picked up by gobld.

**Comment by mark-vieira** [comment]
> It seems that the new images where the logs would be included were not picked up by gobld.

What's indicating this? If this were the case wouldn't we get a completely empty `journalctl.log` file? Checking again today I'm seeing the same in the log.

```
-- Logs begin at Wed 2019-10-30 04:09:46 UTC, end at Wed 2019-10-30 04:42:29 UTC. --
-- No entries --
```

**Comment by mark-vieira** [comment]
Also, I think we should consider blacklisting these images for "general-purpose" builds soon. This is currently the #1 non-test failure cause of build failures. It's getting quite disruptive.

**Comment by mark-vieira** [comment]
> Also, I think we should consider blacklisting these images for "general-purpose" builds soon. This is currently the #1 non-test failure cause of build failures. It's getting quite disruptive.

I take that back, since that wouldn't solve the platform matrix builds 😞

**Comment by atorok** [comment]
The console output would have:
```
Hint: You are currently not seeing messages from other users and the system.
      Users in groups 'adm', 'systemd-journal' can see all messages.
      Pass -q to turn off this notice.
```
The newer images do have `jenkins` added to  `systemd-journal` and manually starting one does show the journal

**Comment by atorok** [comment]
@dliappis we indeed have `overlayfs: lowerdir is in-use as upperdir/workdir` in the logs, but are running a newer docker and older kernel
```
docker version
...
 Engine:
  Version:          19.03.4
...
jenkins@alpar-ci-ubuntu-1804-lts-1030-1809:/dev/shm/elasticsearch$ uname -a 
Linux alpar-ci-ubuntu-1804-lts-1030-1809 5.0.0-1021-gcp #21~18.04.1-Ubuntu SMP Mon Oct 7 04:51:28 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux

```

**Comment by atorok** [comment]
Looking at the failure on Ubuntu: 
```
ct 31, 2019 @ 05:59:55.000
system.syslog
Startup finished in 5.487s (kernel) + 1min 11.772s (userspace) = 1min 17.260s.
Oct 31, 2019 @ 06:00:16.000
system.syslog
2019/10/31 04:00:16 Unsolicited response received on idle HTTP channel starting with "HTTP/1.0 408 Request Time-out\r\nCache-Control: no-cache\r\nConnection: close\r\nContent-Type: text/html\r\n\r\n<html><body><h1>408 Request Time-out</h1>\nYour browser didn't send a complete request in time.\n</body></html>\n"; err=<nil>
Oct 31, 2019 @ 06:01:04.000
system.syslog
link_config: autonegotiation is unset or enabled, the speed and duplex are not writable.
Oct 31, 2019 @ 06:01:04.000
system.syslog
link_config: autonegotiation is unset or enabled, the speed and duplex are not writable.
Oct 31, 2019 @ 06:01:04.000
system.syslog
[  147.260107] audit: audit_lost=8998 audit_rate_limit=500 audit_backlog_limit=8192
Oct 31, 2019 @ 06:01:04.000
system.syslog
time="2019-10-31T04:01:04.832445362Z" level=error msg="error unmounting /var/lib/docker/overlay2/b8c8ed7e60060cd75ce17889efc5ad5fde6a092055fc49a4f089aece59b77c18/merged: invalid argument" storage-driver=overlay2
```

And another: 
```
Oct 31, 2019 @ 10:54:23.000
system.syslog
Startup finished in 6.688s (kernel) + 25.739s (userspace) = 32.427s.
Oct 31, 2019 @ 10:56:12.000
system.syslog
time="2019-10-31T08:56:12.527391244Z" level=error msg="Handler for POST /v1.25/containers/create returned error: error creating overlay mount to /var/lib/docker/overlay2/50846033d2f089f36ae1b22cab94946adfa826beff24fefd0a5d9023904677b5/merged: device or resource busy"
```

Looks like all the failure are ~2 minutes after startup

**Comment by hendrikmuhs** [comment]
another one: https://gradle-enterprise.elastic.co/s/4a2iyzcrcc7ta

**Comment by atorok** [comment]
https://github.com/docker/for-linux/issues/711 seems related but doesn't have any other resolution than using `overlay` instead of `overlay2`

**Comment by atorok** [comment]
I haven't found anything useful for the sles  failures, and it's not clear that switching the storage engine there would help. 
There's no longer a reason to keep these platforms in the general purpose pool. 
We can still get failures but only in the unix compatibility tests so there should be  fewer.

**Comment by mark-vieira** [comment]
Regarding the Ubuntu stuff, looking at that vs our Debian 10 images, everything docker-related looks identical on those two with regards to versions, storage drivers, etc yet we've not seen this on Debian 10. Kernel versions are different though.

I'm not convinced switching storage drivers would help in either case as well.

**Comment by atorok** [comment]
Storage drivers are part in the kernel AFAIK so  they are likely affected by the kernel version.  Also we ran Ubuntu 18 for a long time without issues, so there was likely some update that caused it. 

I tried reproducing the issue on Ubuntu  by bringing test fixtures up and down overnight, but wasn't able to, it ran successfully until the  disk was filled, so that again leads me to think that it could  be something that reproduces close to boot only. 
We are not able to try SLES due to an open issue we can't ssh into it.

**Comment by atorok** [comment]
New failure, this time on Debian 10:
https://gradle-enterprise.elastic.co/s/ccuzjjapt57ok

**Comment by mark-vieira** [comment]
> New failure, this time on Debian 10

Well, that would make sense. Debian 10 is the basis for Ubuntu 18 so it's actually surprising to me we haven't seen a failure until now.

**Comment by mark-vieira** [comment]
So we haven't changed anything that I'm aware of yet these failures have become much less frequent.

![image](https://user-images.githubusercontent.com/4106672/69282144-70987780-0b9e-11ea-8809-922a087b7fe6.png)

**Comment by atorok** [comment]
I did at some point pull some platfroms from the generic pool

**Comment by dliappis** [comment]
Another occurrence on SLES-12 this time in https://gradle-enterprise.elastic.co/s/sh3vsine3eibm

**Comment by polyfractal** [comment]
A SLES-15 failure with remounting tmpfs:

https://gradle-enterprise.elastic.co/s/e6mnydg5e2e42

```
ERROR: for 5929acdd29a349f2cd9a9e375d3a316e_gcs-fixture__gcs-fixture_1  Cannot start service gcs-fixture: unable to remount dir as readonly: mount tmpfs:/var/lib/docker/containers/0fed5381ea84cc6e67bd4ae24d6f0067a97b91f9ca2ff48b4fba1301e24ec980/mounts/secrets, flags: 0x21, data: uid=0,gid=0: device or resource busy
```

**Comment by dliappis** [comment]
Again SLES-15 with remounting tmpfs as mentioned in the [earlier comment](https://github.com/elastic/elasticsearch/issues/48027#issuecomment-561793134):

https://gradle-enterprise.elastic.co/s/zsyg6cijfhzei

**Comment by andreidan** [comment]
Occurred again https://gradle-enterprise.elastic.co/s/j5p6uavcgdfgm

**Comment by pgomulka** [comment]
occurred again, but looks to be more similar to previously mentioned hdfs related failure? https://github.com/elastic/elasticsearch/issues/48027#issuecomment-542684246

https://gradle-enterprise.elastic.co/s/tzshpjeyj5ahu/

**Comment by mark-vieira** [comment]
I'm wondering if this has anything to do with the fact that we are mounting a bind volume which is pointing to a ramdisk backed location. I'm going to setup some experiments to see.

**Comment by mark-vieira** [comment]
Ok, well I got a failure when not running in a `tmpfs` workspace directory so that's not the issue.

https://gradle-enterprise.elastic.co/s/ackcrlydkwlwq

**Comment by mark-vieira** [comment]
My next theory is that somehow bind mounting the same location into multiple containers is causing some kind of weird race condition. My next attempt is going to be to rework the docker compose definitions to instead define this "shared" directory as a named volume, then map that into each container.

**Comment by mark-vieira** [comment]
Ok, so that theory doesn't hold water either since we have encountered failures spinning up fixtures whose compose yaml contains only a single service so contention amongst multiple containers doesn't make sense.

At this point my next theory is simply trying to spin up multiple fixtures concurrently is causing issues. Perhaps introduce some kind of exclusive lock such that we don't run `docker-compose` tasks in parallel? This would introduce a potentially bottleneck in certain builds but I'll trade stability over performance if it's indeed the case.

**Comment by mark-vieira** [comment]
Since Gradle 6.1 introduces the concept of a ["build service"](https://docs.gradle.org/release-candidate/javadoc/org/gradle/api/services/BuildService.html) that would allow us to enforce this concurrency constraint I'm going to defer until that release (which I expect should happen within days).

**Comment by atorok** [comment]
@mark-vieira I would suggest an additional test. Add a 5 minute delay before starting Gradle to the test jobs and see if it reproduces. I was under the impression that some of these failures are happening only after boot, and something might not be initializes yet.

**Comment by mark-vieira** [comment]
That's a reasonably easy thing to test. I'll try that as well. Thanks!

**Comment by astefan** [comment]
Another failure here, on 7.5: https://gradle-enterprise.elastic.co/s/725vwiyumiutq/console-log?task=:test:fixtures:s3-fixture:composeUp

**Comment by polyfractal** [comment]
Looks like we just had another `"error creating overlay mount"` failure :(  Unsure if the delay fix would affect all branches, but this failed on 7.x

https://gradle-enterprise.elastic.co/s/ipms67bf7dwma

```
13:08:03 > Task :test:fixtures:s3-fixture:composeUp FAILED
13:08:03 Building s3-fixture
13:08:03 Building s3-fixture-with-session-token
13:08:03 Building s3-fixture-with-ec2
13:08:03 Building s3-fixture-with-ecs
13:08:03 Creating network "a53b33173db1c753310c1e609ab6be24_s3-fixture__default" with the default driver
13:08:03 Creating a53b33173db1c753310c1e609ab6be24_s3-fixture__s3-fixture-with-ecs_1 ... 
13:08:03 Creating a53b33173db1c753310c1e609ab6be24_s3-fixture__s3-fixture_1          ... 
13:08:03 Creating a53b33173db1c753310c1e609ab6be24_s3-fixture__s3-fixture-with-session-token_1 ... 
13:08:03 Creating a53b33173db1c753310c1e609ab6be24_s3-fixture__s3-fixture-with-ec2_1           ... 
13:08:03 Creating a53b33173db1c753310c1e609ab6be24_s3-fixture__s3-fixture-with-ec2_1           ... error
13:08:03 
13:08:03 ERROR: for a53b33173db1c753310c1e609ab6be24_s3-fixture__s3-fixture-with-ec2_1  Cannot create container for service s3-fixture-with-ec2: error creating overlay mount to /var/lib/docker/overlay2/5802df92173d1947bbde63ec0835daf9d303524fcee1c241463ab8b4755da36f/merged: device or resource busy
13:08:03 Creating a53b33173db1c753310c1e609ab6be24_s3-fixture__s3-fixture-with-ecs_1           ... done
13:08:03 Creating a53b33173db1c753310c1e609ab6be24_s3-fixture__s3-fixture-with-session-token_1 ... done
13:08:03 Creating a53b33173db1c753310c1e609ab6be24_s3-fixture__s3-fixture_1                    ... done
13:08:03 
13:08:03 ERROR: for s3-fixture-with-ec2  Cannot create container for service s3-fixture-with-ec2: error creating overlay mount to /var/lib/docker/overlay2/5802df92173d1947bbde63ec0835daf9d303524fcee1c241463ab8b4755da36f/merged: device or resource busy
13:08:03 Encountered errors while bringing up the project.
```

**Comment by mark-vieira** [comment]
Thanks for that, Zach. Yeah, the delay is introduced for _all_ builds so the theory there was that something on the host wasn't properly initialized and introducing the delay would help.

I think that theory is busted at this point. Also the fact that _some_ of the compose tasks succeed and others do not kinda continues to lean me towards some kind of race condition in concurrently calling `docker-compose`.

I think experimenting with _not_ running any of these tasks in parallel is a worthwhile experiment, even if it's a shitty solution. There's definitely a build performance penalty associated with it.

**Comment by mark-vieira** [comment]
Still seeing failures even after merging https://github.com/elastic/elasticsearch/pull/51439. Starting to run out of ideas here.

**Comment by mark-vieira** [comment]
I still think this might be _somewhat_ related though. The number of occurrences are definitely _less_ in `master` after the "fix" which leads me to believe that there is some kind of race condition but perhaps there are other processes in play here, not just our `composeUp` tasks.

**Comment by mark-vieira** [comment]
Here's something interesting, you can see in this example (which is missing a failure message of any kind) that calling `composeUp` first tries to remove some existing containers. As if the task has already been run, or there is some kind of stale state on the machine. Given this is an ephemeral worker, this makes no sense to me.

https://gradle-enterprise.elastic.co/s/4uw3yxiq6ep4g/console-log?task=:x-pack:test:idp-fixture:composeUp

**Comment by mark-vieira** [comment]
I'm wondering if there's also an issue with interacting with other concurrently running `composeDown` tasks? Perhaps all these should be restricted to be run serially to avoid any funniness. My test build only runs `composeUp` which would explain why I've no longer seen any issues after merging #51439.

**Comment by tlrx** [comment]
It failed today on master: https://gradle-enterprise.elastic.co/s/aj77f6ib2zoy2 

I'm not sure this is the exact same issue, but since a fix was merged a week ago you might find it interesting @mark-vieira :
`
Caused by: java.lang.RuntimeException: Exit-code 1 when calling /usr/local/bin/docker-compose, stdout: N/A
`

**Comment by mark-vieira** [comment]
Yes, I have seen this continue to happen on master as mentioned in https://github.com/elastic/elasticsearch/issues/48027#issuecomment-578854828. The specific error differs between linux distributions but I'm quite certain they are all related, but the root cause evades me.

I'm at my wits-end regarding this issue to be honest and I'm frankly out of ideas. We need to consider simply disabling any tests that rely on docker test fixtures on these known "bad" Linux distributions.

**Comment by mark-vieira** [comment]
That said, it seems since merging my "fix" into master, only SLES remains a culprit here so that might make it easier to justify blacklisting that distro.

**Comment by mark-vieira** [comment]
So it does indeed seem like it's just SLES that's failing with this now after that change. It seems we already blacklist SLES for other Docker-related build steps in https://github.com/elastic/elasticsearch/blob/0a3a7d6179aabe21d1f51a88284fa183b2769d76/.ci/dockerOnLinuxExclusions#L16-L15 so it only makes sense to do so for `docker-compose` as well. We have a lot of Docker-related build logic spread across several plugins, I think it makes sense to finally align this stuff, so that when we "blacklist docker" we do so across the board, not just for some operations. I'm going to do that in a PR to address this issue.

**Comment by davidkyle** [comment]
> So it does indeed seem like it's just SLES that's failing with this now after that change.

Here's a failure for SLES 15 on 6.8. 

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.8+multijob-unix-compatibility/os=sles-15&&immutable/562/console

`
ERROR: for elasticsearch-oss  Cannot start service elasticsearch-oss: unable to remount dir as readonly: mount tmpfs:/var/lib/docker/containers/d4f77774c82c36f495d186c3f3e615d51aaf7176456bd5853cd6c7677c001cca/mounts/secrets, flags: 0x21, data: uid=0,gid=0: device or resource busy
`

**Comment by davidkyle** [comment]
And a release test with the same error 

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+7.6+release-tests/104/console

```
ERROR: for d8865decd592dfc16994307e75357a63_idp-fixture__shibboleth-idp_1  Cannot create container for service shibboleth-idp: error creating overlay mount to /var/lib/docker/overlay2/ed1a5dbf1008faaa838bf73387ba916a20e90ccd43fca7a3bbdc73cea62b4646/merged: device or resource busy
```

**Comment by mark-vieira** [comment]
Grrr, looks like Debian 10 is still having occasional issues as well.

**Comment by pugnascotia** [comment]
Another SLES 15 failure, this time in `:qa:remote-clusters:composeUp` 😢

https://gradle-enterprise.elastic.co/s/ew2jsmzctud2q

**Comment by mark-vieira** [comment]
Fix for this is coming.

**Comment by astefan** [comment]
Another [failure, on 7.6 with SLES 12 for task `:test:fixtures:minio-fixture:composeUp` ](https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+7.6+multijob-unix-compatibility/os=sles-12&&immutable/65/console)with identical error message:

```
01:26:49 > Task :test:fixtures:minio-fixture:composeUp
01:26:49 Creating 2b75b65dda7279078f50b53ea2c3c31f_minio-fixture__minio-fixture_1 ... 
01:26:49 Creating 2b75b65dda7279078f50b53ea2c3c31f_minio-fixture__minio-fixture-for-snapshot-tool_1 ... 
01:26:49 Creating 2b75b65dda7279078f50b53ea2c3c31f_minio-fixture__minio-fixture_1                   ... error
01:26:49 
01:26:49 ERROR: for 2b75b65dda7279078f50b53ea2c3c31f_minio-fixture__minio-fixture_1  Cannot start service minio-fixture: unable to remount dir as readonly: mount tmpfs:/var/lib/docker/containers/932871950a6887081a1b8db70e49b456571fd208b17478d6c150a5072440f7e1/mounts/secrets, flags: 0x21, data: uid=0,gid=0: device or resource busy
01:26:49 Creating 2b75b65dda7279078f50b53ea2c3c31f_minio-fixture__minio-fixture-for-snapshot-tool_1 ... done
01:26:49 
01:26:49 ERROR: for minio-fixture  Cannot start service minio-fixture: unable to remount dir as readonly: mount tmpfs:/var/lib/docker/containers/932871950a6887081a1b8db70e49b456571fd208b17478d6c150a5072440f7e1/mounts/secrets, flags: 0x21, data: uid=0,gid=0: device or resource busy
01:26:49 Encountered errors while bringing up the project.
01:26:49 Stopping 2b75b65dda7279078f50b53ea2c3c31f_minio-fixture__minio-fixture-for-snapshot-tool_1 ... 
```

**Comment by mark-vieira** [comment]
This should be addressed by https://github.com/elastic/elasticsearch/pull/52736 as we'll no longer try to run docker tasks on these known "bad" operating systems.

**Comment by dliappis** [comment]
@mark-vieira while investigating this failure: https://gradle-enterprise.elastic.co/s/5skdsekiggbmm (branch #7.6),
I still see `krb5kdc-fixture` attempting to create docker networks (see below) on `opensuse-15.1`. Based on #42736 and https://github.com/elastic/elasticsearch/blob/bdacaeaf8c7b40ce33950a49e29c651a3eb9d5b1/.ci/dockerOnLinuxExclusions#L9

this task shouldn't execute, any idea what went wrong here? 

```
> Task :test:fixtures:krb5kdc-fixture:composeUp
09:28:47 Building hdfs
09:28:48 /bin/sh: /tmp/_MEIVjssgb/libreadline.so.7: no version information available (required by /bin/sh)
09:28:48 Creating network "8e4ad254157fd4e9add1175f6536a0ad_krb5kdc-fixture__default" with the default driver
09:28:48 Creating 8e4ad254157fd4e9add1175f6536a0ad_krb5kdc-fixture__peppa_1 ... 
09:28:48 Creating 8e4ad254157fd4e9add1175f6536a0ad_krb5kdc-fixture__hdfs_1  ... 
09:28:49 Creating 8e4ad254157fd4e9add1175f6536a0ad_krb5kdc-fixture__peppa_1 ... done
09:28:49 Creating 8e4ad254157fd4e9add1175f6536a0ad_krb5kdc-fixture__hdfs_1  ... done
09:28:49 
09:28:49 > Task :distribution:bwc:maintenance:fetchLatest
09:28:51 
09:28:51 > Task :test:fixtures:krb5kdc-fixture:composeUp
09:28:51 /bin/sh: /tmp/_MEI5Ycyaj/libreadline.so.7: no version information available (required by /bin/sh)
09:28:51 /bin/sh: /tmp/_MEIHZ3YZN/libreadline.so.7: no version information available (required by /bin/sh)
09:28:51 Will use 172.18.0.1 (network 8e4ad254157fd4e9add1175f6536a0ad_krb5kdc-fixture__default) as host of peppa
09:28:51 Will use 172.18.0.1 (network 8e4ad254157fd4e9add1175f6536a0ad_krb5kdc-fixture__default) as host of hdfs
...
```

**Comment by mark-vieira** [comment]
Thanks for reporting this @dliappis. It seems the reported identifier on opensuse has changed slightly. I've updated this in f2c0ad60b16a4a69b5294d00d309c998526281c0 so these docker tasks will be properly skipped on this platform.
